### PR TITLE
feat: add piece moves on current player's turn validation

### PR DESCRIPTION
**Pull request by castawaypirate**
Test's and validator's messages may need to be reviewed.

### CODE REVIEW

**Comment by maliouris** [comment]
[When a piece is going to move validate that its current player's turn](https://trello.com/c/8t2q8B3R/3-when-a-piece-is-going-to-move-validate-that-its-current-players-turn)

File: app/Http/Requests/MovePieceRequest.php [hunk 1]
```diff
@@ -33,6 +33,9 @@ function (Validator $validator) {
                 if (!in_array([$this->request->get('positionX'), $this->request->get('positionY')], $this->piece->availableMoves())) {
                     $validator->errors()->add('position', 'Invalid move');
                 }
+                if ($this->piece->game->playerTurn()->id !== app(CurrentPlayer::class)->get()->id) {
```
Comment (maliouris) [inline app/Http/Requests/MovePieceRequest.php:36]:
```suggestion
                if ($this->piece->game->playerTurn()->id !== $this->piece->player->id) {
```

**Review by maliouris — CHANGES_REQUESTED** [review]

File: tests/Unit/Http/Requests/MovePieceRequestTest.php
Comment (maliouris) [inline tests/Unit/Http/Requests/MovePieceRequestTest.php:1]: Add two properties for white and black piece and remove `$game` property from the class
```php
protected Piece $whitePiece;
protected Piece $blackPiece;
```

In the setup function initialize `$whitePiece` and `$blackPiece` and remove `$blackPiece` initialization from every test

In order for the last test to execute properly after the changes replace `$whitePlayer` with `$blackPlayer`

**Review by maliouris — CHANGES_REQUESTED** [review]

File: app/Models/Piece.php [hunk 2]
```diff
@@ -118,6 +118,11 @@ function move(int $positionX, int $positionY) {
             $this->killEnemy($positionX, $positionY);
         }
 
+        $move = new Move();
+        $move->positionX = $positionX;
+        $move->positionY = $positionY;
+        $this->moves()->save($move);
+
```
Comment (maliouris) [inline app/Models/Piece.php:125]: After the rebase onto main branch you do no longer need those lines
```suggestion
```

**Review by maliouris — CHANGES_REQUESTED** [review]

**Comment by castawaypirate** [comment]
Requested changes have been made. Waiting @maliouris to review.

**Review by maliouris — APPROVED** [review]

**Pull request merged**